### PR TITLE
Claim Contract Income and Close CLI

### PR DESCRIPTION
**Pull request by asamere**
```
➜  arkeo git:(feat/claim-cli) ✗ arkeod arkeo claim --from adam --provider-pubkey tarkeopub1addwnpepq0h7hn9jzhkfwkxgp6kl3ljtjxfvz48emzdrrt5epzjrumpx9kz3w9mjsq9 --nonce 21 --service gaia-mainnet-rpc-archive -y
code: 0
codespace: ""
data: ""
events: []
gas_used: "0"
gas_wanted: "0"
height: "0"
info: ""
logs: []
raw_log: '[]'
timestamp: ""
tx: null
txhash: 40C30A0433BC5D4C19D5D3A718A99464DDE373BE4FA6E76ECC498AE1E8152A35
➜  arkeo git:(feat/claim-cli) ✗ arkeod arkeo claim --from adam
Specify nonce: 51
Specify provider pubkey: tarkeopub1addwnpepq0h7hn9jzhkfwkxgp6kl3ljtjxfvz48emzdrrt5epzjrumpx9kz3w9mjsq9
Specify service (e.g. gaia-mainnet-rpc-archive, btc-mainnet-fullnode, etc): gaia-mainnet-rpc-archive
auth_info:
  fee:
    amount: []
    gas_limit: "200000"
    granter: ""
    payer: ""
  signer_infos: []
  tip: null
body:
  extension_options: []
  memo: ""
  messages:
  - '@type': /arkeo.arkeo.MsgClaimContractIncome
    contract_id: "15"
    creator: tarkeo14q4qnm4tmkm9xuhjwu0vw0f8xy7ztexek44nw3
    nonce: "51"
    signature: zBo3sjVweaiojwFmTGTaCPWDYqTLMFGwycuvf5Yg7bJqCToOm6PCZPbFHh8xN9BP2wLloc8To1DuPukwdbK2Xg==
    spender: tarkeopub1addwnpepq2tjwwcpqwswatymx7uw3q75sqhljmp0qw3pz2fvnavkv7k2jvknqk25q7j
  non_critical_extension_options: []
  timeout_height: "0"
signatures: []
confirm transaction before signing and broadcasting [y/N]: y
code: 0
codespace: ""
data: ""
events: []
gas_used: "0"
gas_wanted: "0"
height: "0"
info: ""
logs: []
raw_log: '[]'
timestamp: ""
tx: null
txhash: B4E6C203AA25957CDB01D3D0CFA6EF12EEBE566A17299CC8FB106009B63BC1FC
```

### CODE REVIEW

File: cmd/directory/indexer/main.go [hunk 1]
```diff
@@ -37,7 +37,6 @@ var (
 		"TENDERMINT_API",
 		"TENDERMINT_WS",
 		"CHAIN_ID",
-		"INDEXER_ID",
```
Comment (asamere) [inline cmd/directory/indexer/main.go:40]: no longer used

File: arkeocli/claim.go [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+	queryCtx, err := client.GetClientQueryContext(cmd)
+	if err != nil {
+		return err
+	}
+
+	queryClient := types.NewQueryClient(queryCtx)
+
+	var contract types.Contract
+	contractID, _ := cmd.Flags().GetUint64("contract-id")
```
Comment (johnnyluo) [inline arkeocli/claim.go:81]: please do not ignore this err

File: arkeocli/claim.go [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+		res, err := queryClient.FetchContract(cmd.Context(), params)
+		if err != nil {
+			return errors.Wrapf(err, "error fetching contract %d", contractID)
+		}
+		if res == nil {
+			return fmt.Errorf("no contract found %d", contractID)
+		}
+		contract = res.GetContract()
+	} else {
+		providerPubkey, _ := cmd.Flags().GetString("provider-pubkey")
```
Comment (johnnyluo) [inline arkeocli/claim.go:93]: please don't ignore this error

File: arkeocli/claim.go [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+		if providerPubkey == "" {
+			providerPubkey, err = promptForArg(cmd, "Specify provider pubkey: ")
+			if err != nil {
+				return
+			}
+		}
+
+		service, _ := cmd.Flags().GetString("service")
```
Comment (johnnyluo) [inline arkeocli/claim.go:101]: same here

**Review by johnnyluo — COMMENTED** [review]
A few comments

File: arkeocli/claim.go (hunk 2 above)
Comment (asamere) [reply arkeocli/claim.go:81]: agreed

**Comment by asamere** [comment]
https://github.com/arkeonetwork/arkeo/issues/130

File: arkeocli/claim.go [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+		clientPubkey = contract.GetClient()
+	}
+
+	creatorAddr, err := clientPubkey.GetMyAddress()
+	if err != nil {
+		return err
+	}
+	creator := creatorAddr.String()
+
+	signStr := fmt.Sprintf("%d:%s:%d", contract.Id, spenderPubkey, nonce)
```
Comment (0xean) [inline arkeocli/claim.go:93]: I believe chad removed spenderPubKey recently.  there is a GetBytesToSign call, which we can reuse for this I think

File: arkeocli/close.go [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+	if err = client.SetCmdClientContext(cmd, clientCtx); err != nil {
+		return
+	}
+
+	contract, err := getContract(cmd)
+	if err != nil {
+		return err
+	}
+
+	if contract.Type == types.ContractType_PAY_AS_YOU_GO {
```
Comment (0xean) [inline arkeocli/close.go:53]: you can remove this check, there are occasions when a client can close a PAY AS YOU GO, for instance when a provider has unbonded.

**Review by 0xean — CHANGES_REQUESTED** [review]

File: arkeocli/claim.go (hunk 5 above)
Comment (asamere) [reply arkeocli/claim.go]: needed a rebase; switched to use types.GetSignBytes(..)

**Review by 0xean — APPROVED** [review]

**Pull request merged**